### PR TITLE
 feat(tornado): support Tornado

### PR DESCRIPTION
**Pull request by ranrib**
1. Added Tornado support.
2. Reduced timeout to 500ms.

### CODE REVIEW

File: epsagon/trace.py [hunk 1]
```diff
@@ -233,18 +233,22 @@ def send_traces(self):
...
+                data=trace,
                 timeout=SEND_TIMEOUT
             )
             if self.debug:
                 print("Sending traces:")
                 pprint.pprint(self.to_dict())
         except requests.exceptions.ReadTimeout as exception:
-            if self.debug:
-                print("Failed to send traces (timeout): ", exception)
+            print("Failed to send trace (size: {}) (timeout): {}".format(
```
Comment (galbash) [inline epsagon/trace.py]: don't print this to the user

Comment (ranrib) [reply epsagon/trace.py]: Done.

File: epsagon/runners/tornado.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                self.resource['metadata'],
+                'Query',
+                request.query
+            )
+
+    def update_response(self, response):
+        """
+        Adds response data to event.
+        :param response: WSGI Response
+        :return: None
```
[the remark this answers is not in the document]
Comment (ranrib) [reply epsagon/runners/tornado.py]: Done.

File: epsagon/runners/tornado.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+from __future__ import absolute_import
+import uuid
+from ..event import BaseEvent
+from ..utils import add_data_if_needed
+
+
+class TornadoRunner(BaseEvent):
+    """
+    Represents Python Flask event runner.
```
[the remark this answers is not in the document]
Comment (ranrib) [reply epsagon/runners/tornado.py]: Done.

File: epsagon/modules/tornado.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        """
+        _, exception, _ = args
+        cls.RUNNER.set_exception(exception, traceback.format_exc())
+        return wrapped(*args, *kwargs)
+
+
+def patch():
+    """
+    Patch module.
+    :return: None
```
[the remark this answers is not in the document]
Comment (ranrib) [reply epsagon/modules/tornado.py]: Done.

File: epsagon/modules/tornado.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        :param instance: wrapt's instance
+        :param args: wrapt's args
+        :param kwargs: wrapt's kwargs
+        :return: None
+        """
+        content = instance._headers.get(  # pylint: disable=protected-access
+            'Content-Type',
+            ''
+        )
+        ignored = ignore_request(content, '')
```
[the remark this answers is not in the document]
Comment (ranrib) [reply epsagon/modules/tornado.py]: There are 2 checks for ignores - one at the request (with the file path) and one for the response with the content type. these are different checks.

[the remark this answers is not in the document]
Comment (ranrib) [reply epsagon/modules/tornado.py]: But anyway found a bug - added `and cls.RUNNER`

File: epsagon/utils.py [hunk 6]
```diff
@@ -18,6 +18,24 @@
     ],
 }
 
+# Ignored content types for web frameworks.
```
[the remark this answers is not in the document]
Comment (ranrib) [reply epsagon/utils.py]: Done.

File: epsagon/runners/tornado.py [hunk 7]
```diff
@@ -29,7 +30,8 @@ def __init__(self, start_time, request):
 
         self.event_id = str(uuid.uuid4())
 
-        self.resource['name'] = 'Tornado'
+        # Since Tornado doesn't has app name, we use the tracer app name.
+        self.resource['name'] = tracer.app_name
```
Comment (galbash) [inline epsagon/runners/tornado.py]: this sounds like a bad idea. is this what we are doing in flask? i think in flask we use the route, thats better (similar to api gateway)

File: epsagon/trace.py [hunk 8]
```diff
@@ -245,10 +245,11 @@ def send_traces(self):
                 print("Sending traces:")
                 pprint.pprint(self.to_dict())
         except requests.exceptions.ReadTimeout as exception:
-            print("Failed to send trace (size: {}) (timeout): {}".format(
-                len(trace),
-                exception
-            ))
+            if self.debug:
+                print("Failed to send trace (size: {}) (timeout): {}".format(
```
Comment (galbash) [inline epsagon/trace.py]: we should really use logging with levels but not for now

File: epsagon/runners/tornado.py (hunk 7 above)
Comment (ranrib) [reply epsagon/runners/tornado.py]: Done.

**Pull request merged**